### ISSUE TITLE
Instance disapeared but files are still there

### ISSUE THREAD

**Issue by Raffle703** [gorilla-devs/GDLauncher#579]
**Describe the bug**
Was playing Sevtech Ages but then the game crashed and the instance disappeared from the GDLauncer selection screen so I looked to see if the files were gone too but they were still there

**Screenshots**
![Instances](https://user-images.githubusercontent.com/68618607/88111548-5cf94280-cb6b-11ea-8a50-313009bdbdcb.jpg)
![Files](https://user-images.githubusercontent.com/68618607/88111551-5f5b9c80-cb6b-11ea-9972-119ee2ac1962.jpg)

**Operating System:**
 - OS: Windows

**Comment by Vexsanity** [comment]
I just got GDLauncher today, downloaded DDSS and came back from dinner to find it also gone from the instance list.

**Comment by rwallac** [comment]
I just had this issue, I installed a new instance of a modpack, I used a different name, and once I closed and re-opened the launcher, it vanished, but the files are all there

**Comment by Ladvace** [comment]
I don't think It crashed because of the Launcher, btw because of the crash the config seems to be corrupted that's why you can't see it

**Comment by blarfoon** [comment]
If anyone still has this issue and did not delete the folder, please contact me on discord `Davide#1234`

**Comment by ScoopTwo** [comment]
had the same issue i renamed the file and it showed up immediately in gdlaunch
![2020-08-11 (8)](https://user-images.githubusercontent.com/57631417/89945012-b4308700-dbee-11ea-8ddc-ffd28d500083.png)
er

**Comment by imNaphy** [comment]
Happened to me too, this needs to be fixed

Edit: forgot to mention, also fixed by renaming the folder of the modpack

**Comment by DiefBell** [comment]
I'm getting this issue too and renaming the folder of the modpack is not fixing it

**Comment by blarfoon** [comment]
This should be fixed now

**Comment by DiefBell** [comment]
@killpowa seems to be working for me, thanks. What was the issue out of interest?

**Comment by blarfoon** [comment]
Curseforge provided a result for the mod, but when we actually asked for data it would return 404 and the launcher was confused

**Comment by Sanguinely** [comment]
Has happened again, renaming did not fix

**Comment by blarfoon** [comment]
Could you contact me on discord Davide#1234 ?

**Comment by Sanguinely** [comment]
@killpowa I was just making a post to say that it still is happening, my friend had doubleclicked on Enigmatica 6 modpack and it opened 2 instances and then crashed because of ram overload. that's about as much info as i can give. was fixxed by deleting the files and reinstalling the pack

sorry if it's not much help

**Comment by mkolassa** [comment]
I just had this happen as well.  I had a graphics driver lockup while both Minecraft and GDLauncher were running, and my instance disappeared.  The config.json was full of NULL characters.

I wonder if writing a backup config.json wouldn't be advisable?

**Comment by blarfoon** [comment]
@mkolassa could you contact me on discord?

**Comment by mkolassa** [comment]
> @mkolassa could you contact me on discord?

Looks like I need to add you as a friend to message you.  I have sent you a friend request.

**Comment by GooberMcNugget** [comment]
Hey I've had this issue happen a couple of times with the same modpack (proejct ozone 3) and changing the name didnt fix it either time. The first time I ended up redownloading the pack, I haven't fixed it this time around. I think my computer was forced to shut down (we love siblings that press and hold the power button to turn off computers!) while the game was running. Could that have something to do with it?

Edit: Just replaced the config.json file in the main folder with a copy of one from a fresh download of the pack, restarted the launcher, and it showed up again. The old config file wasn't readable by VSCode...

**Comment by mkolassa** [comment]
I ended up doing the same thing.  I was worried that I'd have an issue with some of the mods that I've added to the pack, but it picked them up without issue.

**Comment by magicmanuk** [comment]
I have another instance of this issue - MineColonies Official dropped off my launcher (Discord contact sent)
Launch button gone, instance files remain. I've just re-installed the pack with different name, closed & reopened GDL & that button is still there
Edit: Looks like my config.json also got corrupted and copying the one from the fresh DL , restarting GDL has got that instance launch button back

**Comment by Irgendwer01** [comment]
Happened me tho, I opened the config.json with Notepad++ and seen that anything got overwritten with NUL

**Comment by Lennster1** [comment]
what does that mean? Having same issue

**Comment by ClaymooreEOC** [comment]
Same issue here. Had a crash and the All of Fabric pack is gone.
Where can i find that config.json file?

**Comment by blarfoon** [comment]
Crash of your pc/instance or launcher?

**Comment by ClaymooreEOC** [comment]
The PC crashed. Complete black screen.
First time ive seen that on win10.
Had to do a hard reset.

**Comment by mdh196884** [comment]
I lost power while playing today, and when I was able to open the launcher again, the instance I was playing was gone.

As others have pointed out, the files are still there, except the `config.json` file seems to have been corrupted. It has approximately the same length as a fresh `config.json` file from a new instance of the same pack, but it contains nothing but `NUL` characters.

I renamed the corrupt `config.json` file and moved in a copy of the `config.json` from a new instance. The pack now appears in the launcher and works fine, but it looks like I've lost my recorded playtime. It might not be a perfect fix (I don't even know what info is stored in `config.json` myself), but it got my instance working again.

**Comment by guywbailey** [comment]
I added three mods to a mod folder. Then, I went to run GD Launcher 1.0.11, and it said the file wasn't there anymore.

**Comment by blarfoon** [comment]
Hopefully fixed by https://github.com/gorilla-devs/GDLauncher/commit/c5c3c6ad4c263bb19f74e332598d444be3ff6c04

**Comment by skorpeyon** [comment]
This happened to me today, but without a crash, after simply installing a new instance, then exiting the game and launcher. Next time I opened GDLauncher an older instance had disappeared. Attempted to edit the name of the folder, did not return. The config.json file has disappeared in that folder. I did attempt to copy it over from the only other instance I have, but the information is incorrect (different MC versions, etc.). Should be able to recover it by reinstalling the modpack that instance was built from (added a lot of additional mods), then copying over the config.json from the new modpack, I suppose, but wanted to point out that this appears to still be ongoing.
